### PR TITLE
fix for odd-even bug in random

### PR DESCRIPTION
**Pull request by martinbaker**
Fix for the following bug and also add comments.

When I call (rndInt 0 100) repeatedly the result does not look very psudorandom, for instance, it gives alternating odd and even numbers.

When I look at the Idris code here:
https://github.com/idris-lang/Idris-dev/blob/master/libs/effects/Effect/Random.idr
and I enter the code manually I also get alternating odd and even numbers:

```
Idris> :let seed = 1
Idris> :let seed=((1664525 * seed + 1013904223) `prim__sremBigInt` (pow 2 32))
Idris> seed
1015568748 : Integer
Idris> :let seed=((1664525 * seed + 1013904223) `prim__sremBigInt` (pow 2 32))
Idris> seed
1586005467 : Integer
Idris> :let seed=((1664525 * seed + 1013904223) `prim__sremBigInt` (pow 2 32))
Idris> seed
2165703038 : Integer
Idris> :let seed=((1664525 * seed + 1013904223) `prim__sremBigInt` (pow 2 32))
Idris> seed
3027450565 : Integer
Idris> :let seed=((1664525 * seed + 1013904223) `prim__sremBigInt` (pow 2 32))
Idris> seed
217083232 : Integer
Idris> :let seed=((1664525 * seed + 1013904223) `prim__sremBigInt` (pow 2 32))
Idris> seed
1587069247 : Integer
Idris> :let seed=((1664525 * seed + 1013904223) `prim__sremBigInt` (pow 2 32))
Idris> seed
3327581586 : Integer
```

### CODE REVIEW

File: libs/effects/Effect/Random.idr [hunk 1]
```diff
@@ -33,7 +48,8 @@ rndFin k = do let v = assert_total $ !(call GetRandom) `prim__sremBigInt` (cast
        toFin x = case integerToFin x (S k) of
                       Just v => v
                       Nothing => toFin (assert_smaller x (x - cast (S k)))
-
+       -- TODO insert divBigInt x 2 to prevent low bit alternating between
```
Comment (ahmadsalim) [inline libs/effects/Effect/Random.idr]: Please remove TODO

**Comment by ahmadsalim** [comment]
@martinbaker It seems some tests are broken, can you please take a look at them?

**Comment by martinbaker** [comment]
test/effects001
test/effects003

are hardcoded to expect particular outputs from the pseudorandom generator.
Since this has changed the tests have to be modified to expect the new values.

I have changed the tests to what I think they should be but I don't really know much about AppVeyor and Travis so I might have got it wrong.

Martin

**Comment by ahmadsalim** [comment]
@martinbaker It seems that `effect003` still fails, but now on a more minor thing.

You can try to run the test locally (by using the script inside the folder).

**Comment by ahmadsalim** [comment]
Thanks!

**Pull request merged**